### PR TITLE
unittest: Handle op_db to align xfail defined by CUDA

### PR DESCRIPTION
**Pull request by yuchengliu1**
Functional coverage of PyTorch XPU operator is aligned with CUDA. So we are gradually align test scope with CUDA. The PR is to handle op_db to align xfail configuration. It helps us reduce efforts to triage failures which are expected.
#593

### CODE REVIEW

**Comment by fengyuan14** [comment]
Please resolve the conflicts.

**Comment by fengyuan14** [comment]
Please,
1. Address the failure of extended case script
2. Resolve the conflicts.

File: test/xpu/extended/run_test_with_skip.py [hunk 1]
```diff
@@ -179,25 +156,26 @@
     "test_compare_cpu_std_mean_xpu_bfloat16",
     "test_compare_cpu_sub_xpu_float16",
     "test_compare_cpu_var_mean_xpu_bfloat16",
-
+    # Numeric difference
+    # Mismatched elements: 2 / 125 (1.6%)
+    # Greatest absolute difference: 0.001953125 at index (2, 0, 0) (up to 0.001 allowed)
+    # Greatest relative difference: 0.007568359375 at index (2, 0, 0) (up to 0.001 allowed)
+    "test_compare_cpu_cumprod_xpu_bfloat16",
```
Comment (fengyuan14) [inline test/xpu/extended/run_test_with_skip.py:163]: We have got it in line 172. Please remove it.

File: test/xpu/run_test_with_skip.py [hunk 2]
```diff
@@ -755,35 +743,28 @@ def launch_test(test_case, skip_list=None, exe_list=None):
...
-    # Greatest relative difference: 0.15330001655652495 at index (765, 860) (up to 1e-07 allowed)
-    "test_python_ref__refs_log2_xpu_complex128",
-
-    #AssertionError: The supported dtypes for unique_consecutive on device type xpu are incorrect!
-    #The following dtypes worked in forward but are not listed by the OpInfo: {torch.bfloat16}.
-    #XPU supports bfloat16, CUDA doesn't support it.
+    # Mismatched elements: 2 / 125 (1.6%)
+    # Greatest absolute difference: 0.001953125 at index (2, 0, 0) (up to 0.001 allowed)
+    # Greatest relative difference: 0.007568359375 at index (2, 0, 0) (up to 0.001 allowed)
+    "test_compare_cpu_cumprod_xpu_bfloat16",
```
Comment (fengyuan14) [inline test/xpu/run_test_with_skip.py:756]: Why we pass before?

**Comment by fengyuan14** [comment]
Please address the comments.

File: test/xpu/extended/run_test_with_skip.py (hunk 1 above)
Comment (yuchengliu1) [reply test/xpu/extended/run_test_with_skip.py]: Done

File: test/xpu/run_test_with_skip.py (hunk 2 above)
Comment (yuchengliu1) [reply test/xpu/run_test_with_skip.py]: removed it

**Review by fengyuan14 — APPROVED** [review]

**Pull request merged**